dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -373,80 +370,9 @@ private static async Task<T> RunRequestAsync<T>(Func<Task<T>> request)
         {
             return await request.Invoke().ConfigureAwait(false);
         }
-        catch (RequestFailedException e)
+        catch (RequestFailedException ex)
         {
-            switch (e.Status)
```

Comment:
The idea here was, by grouping responses into these categories, you could programmatically take action based on whether it was a retriable or intermittent issue, or slow down if it was a throttling issue, etc.
Do we still have that somewhere? (The PR is huge and keeps freezing my browser)

Reply:
I'm going to have to pick up the review later from a desktop. My laptop can't seem to handle the code diff without freezing.